contracts/modules/Experimental/TransferManager/LockupVolumeRestrictionTM.sol

```diff
@@ -229,164 +282,104 @@ contract LockupVolumeRestrictionTM is ITransferManager {
+        for (uint256 i = 0; i < userLockup.length; i++) {
+            // Find out the remaining locked amount for a given lockup
+            uint256 remainingLockedAmount = userLockup[i].lockupAmount.sub(_getUnlockedAmountForLockup(userLockup, i));
+            // aggregating all the remaining locked amount for all the lockups for a given address
+            totalRemainingLockedAmount = totalRemainingLockedAmount.add(remainingLockedAmount);
         }
-
-        return _checkIfUnlockedTokenTransferIsPossible(userAddress, amount, tokenSums[1], tokenSums[2]);
-    }
-
-    function _checkIfUnlockedTokenTransferIsPossible(
-        address userAddress,
-        uint amount,
-        uint totalSum,
-        uint alreadyWithdrawnSum
-        ) internal view returns (Result) {
-        // the amount the user wants to withdraw is greater than their allowed amounts according to the lockups.  however, if the user has like, 10 tokens, but only 4 are locked up, we should let the transfer go through for those 6 that aren't locked up
-        uint currentUserBalance = ISecurityToken(securityToken).balanceOf(userAddress);
-        uint stillLockedAmount = totalSum.sub(alreadyWithdrawnSum);
-        if (currentUserBalance >= stillLockedAmount && amount <= currentUserBalance.sub(stillLockedAmount)) {
-            // the user has more tokens in their balance than are actually locked up.  they should be allowed to withdraw the difference
+        // Present balance of the user
+        uint256 currentBalance = ISecurityToken(securityToken).balanceOf(_userAddress);
+        if ((currentBalance.sub(_amount)) >= totalRemainingLockedAmount) {
             return Result.VALID;
         }
         return Result.INVALID;
     }
 
+    /**
+     * @notice Provide the unlock amount for the given lockup for a particular user
+     */
+    function _getUnlockedAmountForLockup(LockUp[] userLockup, uint256 _lockupIndex) internal view returns (uint256) {
+            /*solium-disable-next-line security/no-block-members*/
+            if (userLockup[_lockupIndex].startTime > now) {
+                return 0;
+            } else {
+                // Calculate the no. of periods for a lockup 
+                uint256 noOfPeriods = (userLockup[_lockupIndex].lockUpPeriodSeconds).div(userLockup[_lockupIndex].releaseFrequencySeconds);
+                // Calculate the transaction time lies in which period
+                /*solium-disable-next-line security/no-block-members*/
+                uint256 elapsedPeriod = (now.sub(userLockup[_lockupIndex].startTime)).div(userLockup[_lockupIndex].releaseFrequencySeconds);
+                // Calculate the allowed unlocked amount per period
+                uint256 amountPerPeriod = (userLockup[_lockupIndex].lockupAmount).div(noOfPeriods);
+                // Find out the unlocked amount for a given lockup
+                uint256 unLockedAmount = elapsedPeriod.mul(amountPerPeriod);
+                return unLockedAmount;
```

Comment:
Rounding off can cause tokens to be locked forever. Consider this example:
Granularity of token = 1
Total locked amount = 10
Number of periods = 6

This means, the amount per lock period = 10/6 = 1.67 = 1.
unlocked amount even after all periods are passed = 6*1 = 1

4 is a very small number as 1 token is 10**18 but it can cause amount mismatch and confusion as it will make transactions revert when use will try to transfer all of locked tokens.